server/storage/wal/wal_test.go

```diff
@@ -368,48 +369,56 @@ func TestSaveWithCut(t *testing.T) {
 	}
 }
 
-func TestRecover(t *testing.T) {
-	p := t.TempDir()
+func TestRecoverWithDifferentEntrySizes(t *testing.T) {
+	zeroMb := 0
+	oneMb := 1 * 1024 * 1024
+	twentyMb := 20 * 1024 * 1024
+	for _, entrySize := range []int{zeroMb, oneMb, twentyMb} {
```

Comment:
Please use subtest instead.  There are lots of examples, such as [discovery_test.go#L66](https://github.com/etcd-io/etcd/blob/b7be91f98b7a7a6202ed9e9bc37b692d484585a8/server/etcdserver/api/v3discovery/discovery_test.go#L66)